"embulk run -I" command doesn't works with v0.9.3+

I noticed `embulk run -I /path/to/embulk-input-gcs/lib /path/to/config.yml" got following failure.
I'm using this command while developing Java based plugins.
This command had been working until Embulk v0.9.2 and but not working since v0.9.3.

There're only 2 PRs between v0.9.2 and v0.9.3. I guess one or both of them affects to this problem.
https://github.com/embulk/embulk/pull/962
https://github.com/embulk/embulk/pull/963

```
$ ./embulk-0.9.4.jar run -I /path/to/embulk-input-gcs/lib gcs_stdout.yml
2018-03-30 15:13:52.724 +0900: Embulk v0.9.4

********************************** INFORMATION **********************************
  Join us! Embulk-announce mailing list is up for IMPORTANT announcement such as
    compatibility-breaking changes and key feature updates.
  https://groups.google.com/forum/#!forum/embulk-announce
*********************************************************************************
2018-03-30 15:13:58.645 +0900 [INFO] (main): Started Embulk v0.9.4
2018-03-30 15:14:06.921 +0900 [INFO] (0001:transaction): Gem's home and path are set by default: "/Users/akama/.embulk/lib/gems"
LoadError: load error: embulk/input/gcs -- java.lang.ClassNotFoundException: org.embulk.input.gcs.GcsFileInputPlugin
           require at org/jruby/RubyKernel.java:955
           require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55
  require_and_show at uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin_registry.rb:81
            search at uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin_registry.rb:39
            lookup at uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin_registry.rb:26
            lookup at uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin.rb:197
    new_java_input at uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin.rb:155
    new_java_input at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/forwardable.rb:189
org.embulk.config.ConfigException: InputPlugin 'gcs' is not found.
(LoadError) load error: embulk/input/gcs -- java.lang.ClassNotFoundException: org.embulk.input.gcs.GcsFileInputPlugin
	at org.embulk.plugin.PluginManager.buildPluginNotFoundException(PluginManager.java:78)
	at org.embulk.plugin.PluginManager.newPluginWithoutWrapper(PluginManager.java:64)
	at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:31)
	at org.embulk.spi.ExecSession.newPlugin(ExecSession.java:147)
	at org.embulk.spi.Exec.newPlugin(Exec.java:63)
	at org.embulk.exec.BulkLoader$ProcessPluginSet.<init>(BulkLoader.java:409)
	at org.embulk.exec.BulkLoader.doRun(BulkLoader.java:484)
	at org.embulk.exec.BulkLoader.access$000(BulkLoader.java:35)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:353)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:350)
	at org.embulk.spi.Exec.doWith(Exec.java:22)
	at org.embulk.exec.BulkLoader.run(BulkLoader.java:350)
	at org.embulk.EmbulkEmbed.run(EmbulkEmbed.java:162)
	at org.embulk.EmbulkRunner.runInternal(EmbulkRunner.java:292)
	at org.embulk.EmbulkRunner.run(EmbulkRunner.java:156)
	at org.embulk.cli.EmbulkRun.runSubcommand(EmbulkRun.java:437)
	at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:92)
	at org.embulk.cli.Main.main(Main.java:26)
	Suppressed: org.embulk.plugin.PluginSourceNotMatchException
		at org.embulk.plugin.InjectedPluginSource.newPlugin(InjectedPluginSource.java:53)
		at org.embulk.plugin.PluginManager.newPluginWithoutWrapper(PluginManager.java:52)
		... 16 more
	Suppressed: org.embulk.plugin.PluginSourceNotMatchException
		at org.embulk.plugin.maven.MavenPluginSource.newPlugin(MavenPluginSource.java:65)
		at org.embulk.plugin.PluginManager.newPluginWithoutWrapper(PluginManager.java:52)
		... 16 more
	Suppressed: org.embulk.plugin.PluginSourceNotMatchException: org.jruby.exceptions.RaiseException: (LoadError) load error: embulk/input/gcs -- java.lang.ClassNotFoundException: org.embulk.input.gcs.GcsFileInputPlugin
		at org.embulk.jruby.JRubyPluginSource.newPlugin(JRubyPluginSource.java:63)
		at org.embulk.plugin.PluginManager.newPluginWithoutWrapper(PluginManager.java:59)
		... 16 more
	Caused by: org.jruby.exceptions.RaiseException: (LoadError) load error: embulk/input/gcs -- java.lang.ClassNotFoundException: org.embulk.input.gcs.GcsFileInputPlugin
		at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:955)
		at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.core_ext.kernel_require.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55)
		at RUBY.require_and_show(uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin_registry.rb:81)
		at RUBY.search(uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin_registry.rb:39)
		at RUBY.lookup(uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin_registry.rb:26)
		at RUBY.lookup(uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin.rb:197)
		at RUBY.new_java_input(uri:classloader:/gems/embulk-0.9.4-java/lib/embulk/plugin.rb:155)
		at RUBY.new_java_input(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/forwardable.rb:189)

Error: InputPlugin 'gcs' is not found.
(LoadError) load error: embulk/input/gcs -- java.lang.ClassNotFoundException: org.embulk.input.gcs.GcsFileInputPlugin

```

--- a/embulk-core/src/main/java/org/embulk/jruby/JRubyInitializer.java
+++ b/embulk-core/src/main/java/org/embulk/jruby/JRubyInitializer.java
@@ -19,16 +19,16 @@ public final class JRubyInitializer {
             final Logger logger,
             final String gemHome,
             final boolean useDefaultEmbulkGemHome,
-            final List<String> jrubyClasspath,
             final List<String> jrubyLoadPath,
+            final List<String> jrubyClasspath,
             final List<String> jrubyOptions,
             final String jrubyBundlerPluginSourceDirectory) {
         this.injector = injector;
         this.logger = logger;
         this.gemHome = gemHome;
         this.useDefaultEmbulkGemHome = useDefaultEmbulkGemHome;
-        this.jrubyClasspath = jrubyClasspath;
         this.jrubyLoadPath = jrubyLoadPath;
+        this.jrubyClasspath = jrubyClasspath;
         this.jrubyOptions = jrubyOptions;
         this.jrubyBundlerPluginSourceDirectory = jrubyBundlerPluginSourceDirectory;
     }
@@ -137,6 +137,32 @@ public final class JRubyInitializer {
                              jruby.runScriptlet("Embulk::Logger"),
                              "new",
                              injector.getInstance(ILoggerFactory.class).getLogger("ruby")));
+    }
+
+    // TODO: Remove these probing methods, and test through mocked ScriptingContainerDelegate.
+
+    String probeGemHomeForTesting() {
+        return this.gemHome;
+    }
+
+    boolean probeUseDefaultEmbulkGemHomeForTesting() {
+        return this.useDefaultEmbulkGemHome;
+    }
+
+    List<String> probeJRubyLoadPathForTesting() {
+        return this.jrubyLoadPath;
+    }
+
+    List<String> probeJRubyClasspathForTesting() {
+        return this.jrubyClasspath;
+    }
+
+    List<String> probeJRubyOptionsForTesting() {
+        return this.jrubyOptions;
+    }
+
+    String probeJRubyBundlerPluginSourceDirectoryForTesting() {
+        return this.jrubyBundlerPluginSourceDirectory;
     }
 
     private void setGemVariables(final ScriptingContainerDelegate jruby) {
@@ -256,8 +282,8 @@ public final class JRubyInitializer {
     private final Logger logger;
     private final String gemHome;
     private final boolean useDefaultEmbulkGemHome;
-    private final List<String> jrubyClasspath;
     private final List<String> jrubyLoadPath;
+    private final List<String> jrubyClasspath;
     private final List<String> jrubyOptions;
     private final String jrubyBundlerPluginSourceDirectory;
 }

--- a/embulk-core/src/test/java/org/embulk/jruby/TestJRubyInitializer.java
+++ b/embulk-core/src/test/java/org/embulk/jruby/TestJRubyInitializer.java
@@ -1,0 +1,49 @@
+package org.embulk.jruby;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+public class TestJRubyInitializer {
+    @Test
+    public void testArguments() {
+        final List<String> loadPath = new ArrayList<>();
+        loadPath.add("/load/path");
+        final List<String> classpath = new ArrayList<>();
+        classpath.add("/classpath");
+        final List<String> options = new ArrayList<>();
+        options.add("--option");
+        options.add("arg");
+
+        final JRubyInitializer initializer = JRubyInitializer.of(
+                null,
+                LoggerFactory.getLogger(TestJRubyInitializer.class),
+                "/gem/home",
+                true,
+                loadPath,
+                classpath,
+                options,
+                "/bundle");
+
+        // TODO: Test through mocked ScriptingContainerDelegate, not through probing methods for testing.
+
+        assertEquals("/gem/home", initializer.probeGemHomeForTesting());
+        assertTrue(initializer.probeUseDefaultEmbulkGemHomeForTesting());
+
+        assertEquals(1, initializer.probeJRubyLoadPathForTesting().size());
+        assertEquals("/load/path", initializer.probeJRubyLoadPathForTesting().get(0));
+
+        assertEquals(1, initializer.probeJRubyClasspathForTesting().size());
+        assertEquals("/classpath", initializer.probeJRubyClasspathForTesting().get(0));
+
+        assertEquals(2, initializer.probeJRubyOptionsForTesting().size());
+        assertEquals("--option", initializer.probeJRubyOptionsForTesting().get(0));
+        assertEquals("arg", initializer.probeJRubyOptionsForTesting().get(1));
+
+        assertEquals("/bundle", initializer.probeJRubyBundlerPluginSourceDirectoryForTesting());
+    }
+}

